### PR TITLE
[TGBOT-25] Create Engineer repository & Bean using Arrays, CRUD and Tests

### PR DESCRIPTION
**Pull request by schneidervs**
https://schneidervs.atlassian.net/browse/TGBOT-25

### CODE REVIEW

**Comment by schneidervs** [comment]
didn't add methods

File: demo/src/main/java/com/example/demo/bean/Engineer.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                ", firstName='" + firstName + '\'' +
+                ", lastName='" + lastName + '\'' +
+                ", function='" + function + '\'' +
+                ", category='" + category + '\'' +
+                ", education='" + education + '\'' +
+                ", experience=" + experience +
+                ", generalExperience=" + generalExperience +
+                '}';
+    }
+}
```
Comment (AlexanderYarokhovich) [inline demo/src/main/java/com/example/demo/bean/Engineer.java:119]: add new line in the end.

**Review by AlexanderYarokhovich — APPROVED** [review]

File: demo/src/main/java/com/example/demo/repository/EngineerRepository.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                    Engineer[] engineersTemp = Arrays.copyOf(engineers, (engineers.length + 5));
+                    engineers = engineersTemp;
+                }
+                break;
+            }
+        }
+        return engineers;
+    }
+
+    public Engineer readEngineer(int i) {
```
Comment (AlexanderYarokhovich) [inline demo/src/main/java/com/example/demo/repository/EngineerRepository.java]: method name "read" , change to "find"

File: demo/src/main/java/com/example/demo/repository/EngineerRepository.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                engineers[j] = null;
+            }
+        }
+        return engineers;
+    }
+
+    public String count(int i) {
+        i = i + 1;
+        String count = String.valueOf(i);
+        return count;
```
Comment (AlexanderYarokhovich) [inline demo/src/main/java/com/example/demo/repository/EngineerRepository.java]: Maybe just write: engineers.length

File: demo/src/main/java/com/example/demo/repository/EngineerRepository.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    }
+}
```
Comment (AlexanderYarokhovich) [inline demo/src/main/java/com/example/demo/repository/EngineerRepository.java]: Add an empty string and methods that are listed in the job

File: demo/src/test/java/com/example/demo/repository/EngineerRepositoryTest.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    }
+
+    void createFive() {
+        engineerRepository.createEngineer(engineer1i);
+        engineerRepository.createEngineer(engineer2i);
+        engineerRepository.createEngineer(engineer3i);
+        engineerRepository.createEngineer(engineer4i);
+        engineerRepository.createEngineer(engineer5i);
+    }
+}
```
Comment (AlexanderYarokhovich) [inline demo/src/test/java/com/example/demo/repository/EngineerRepositoryTest.java]: Add an empty string

File: demo/src/test/java/com/example/demo/repository/EngineerRepositoryTest.java [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    private Engineer engineer5i;
+    private Engineer engineer5o = new Engineer("5", "Нестор", "Майко", "Инжинер по гарантии", "высшая", "высшее", 5, 10);
+    private Engineer engineer4o = new Engineer("4", "Денис", "Пелля", "Мастер цеха", "высшая", "высшее", 3, 7);
+    private Engineer engineer3o = new Engineer("3", "Юрий", "Яковлев", "Главный инжинер", "высшая", "высшее", 1, 10);
+    private Engineer engineer2o = new Engineer("2", "Роман", "Валюшко", "Инжинер по гарантии", "высшая", "высшее", 5, 10);
+    private Engineer engineer1o = new Engineer("1", "Роман", "Валюшко", "Главный менеджер", "высшая", "высшее", 5, 10);
+    private final Engineer engineer5d = new Engineer("4", "Роман", "Валюшко", "Главный менеджер", "высшая", "высшее", 5, 10);
+
+    @BeforeEach
+    public void before() {
```
Comment (AlexanderYarokhovich) [inline demo/src/test/java/com/example/demo/repository/EngineerRepositoryTest.java]: Maybe add a method:
public void setup()

File: demo/src/main/java/com/example/demo/bean/Engineer.java (hunk 1 above)
Comment (schneidervs) [reply demo/src/main/java/com/example/demo/bean/Engineer.java:119]: Look closely, they are present

File: demo/src/main/java/com/example/demo/repository/EngineerRepository.java (hunk 2 above)
Comment (schneidervs) [reply demo/src/main/java/com/example/demo/repository/EngineerRepository.java]: Look carefully, I already did this fix
https://github.com/Tutritp01/TgBot/pull/16/commits/13e2e27f764dc13150f3da84d06b5254be035645

File: demo/src/main/java/com/example/demo/repository/EngineerRepository.java (hunk 3 above)
Comment (schneidervs) [reply demo/src/main/java/com/example/demo/repository/EngineerRepository.java]: It's just a translation from a string value to an int.
please see the fix
https://github.com/Tutritp01/TgBot/pull/16/commits/13e2e27f764dc13150f3da84d06b5254be035645

File: demo/src/main/java/com/example/demo/repository/EngineerRepository.java (hunk 4 above)
Comment (schneidervs) [reply demo/src/main/java/com/example/demo/repository/EngineerRepository.java:102]: please see the fix

File: demo/src/test/java/com/example/demo/repository/EngineerRepositoryTest.java (hunk 5 above)
Comment (schneidervs) [reply demo/src/test/java/com/example/demo/repository/EngineerRepositoryTest.java]: Add an empty string

File: demo/src/test/java/com/example/demo/repository/EngineerRepositoryTest.java (hunk 6 above)
Comment (schneidervs) [reply demo/src/test/java/com/example/demo/repository/EngineerRepositoryTest.java:36]: Why is "public void setup()" better than "public void before()"?

**Comment by schneidervs** [comment]
please see the code in this link
https://github.com/Tutritp01/TgBot/commit/13e2e27f764dc13150f3da84d06b5254be035645

**Pull request merged**